Features/ManageabilityPkg/Library/ManageabilityTransportSsifLib/Dxe/ManageabilityTransportSsif.c

```diff
@@ -362,21 +362,22 @@ ReleaseTransportSession (
   EFI_STATUS                    Status;
   MANAGEABILITY_TRANSPORT_SSIF  *SsifTransportToken;
 
+  Status = EFI_INVALID_PARAMETER;
+
   if (TransportToken == NULL) {
-    Status = EFI_INVALID_PARAMETER;
+    return Status;
   }
 
   SsifTransportToken = MANAGEABILITY_TRANSPORT_SSIF_FROM_LINK (TransportToken);
   if (mSingleSessionToken != SsifTransportToken) {
-    Status = EFI_INVALID_PARAMETER;
+    Status              = EFI_SUCCESS;
+    mSingleSessionToken = NULL;
   }
 
   if (SsifTransportToken != NULL) {
     FreePool (SsifTransportToken->Token.Transport->Function.Version1_0);
     FreePool (SsifTransportToken->Token.Transport);
     FreePool (SsifTransportToken);
-    mSingleSessionToken = NULL;
-    Status              = EFI_SUCCESS;

```

Comment:
I want to confirm with you is this removal correct?

Reply:
Oh. The condition was not changed in this module.
I'll fix it.

This was moved under the condition 
  if (mSingleSessionToken == SsifTransportToken) {
     Status              = EFI_SUCCESS;
     mSingleSessionToken = NULL;
   }
  
Actually I have another version of this changes, but I didn't published it, because it mostly the same, but the patch got bigger. I've added addition check that SsifTransportToken != NULL.
It's a minor issue, because it could be NULL only if sizeof (UINTN) value was passed to function as pointer parameter.

I tried to fix original tricky code with a minimal changes.

Reply:
@changab Please review.